.github/setup-node/action.yml

```
@@ -32,8 +32,15 @@ runs:
 
         - name: Install npm dependencies
           if: ${{ steps.cache-node_modules.outputs.cache-hit != 'true' }}
-          run: npm ci
+          run: |
+              npm ci
           shell: bash
+        - name: Upload npm logs as an artifact on failure
+          uses: actions/upload-artifact@65462800fd760344b1a7b4382951275a0abb4808 # v4.3.3
+          if: failure()
+          with:
+              name: npm-logs
```

I think we'll need to add something else to the name. Otherwise the jobs will overwrite each other, or only the first one will succeed. This will append the job's ID number to the artifact name.

```suggestion
              name: npm-logs-${{ github.job }}
```

---

Totally agree! Since you're working on this, feel free to apply this suggestion in your PR (#61289).